apps/vscode-wing/src/extension.ts

```diff
@@ -21,11 +26,62 @@ export async function activate(context: ExtensionContext) {
 }
 
 async function startLanguageServer(context: ExtensionContext) {
-  // Allow the user to override the path, otherwise just use npx
-  const wingBin = env.WING_BIN ?? "npx";
-  const args = env.WING_BIN
-    ? ["lsp"]
-    : ["-y", "-q", `winglang@${context.extension.packageJSON.version}`, "lsp"];
+  const extVersion = context.extension.packageJSON.version;
+  const configuredWingBin = workspace
+    .getConfiguration(CFG_WING)
+    .get<string>(CFG_WING_BIN, "wing");
+  let wingBin = env.WING_BIN ?? configuredWingBin;
+
+  if (wingBin !== "npx") {
+    // check if wing is installed
+    const result = which.sync(wingBin, { nothrow: true });
```

Comment:
I'd consider extracting this one line into a function like `assertWingIsInstalled()` to avoid the need for the comment.